ouroboros-network-framework/src/Simulation/Network/Snocket.hs

```diff
@@ -331,7 +344,10 @@ newNetworkState bearerInfoScript = atomically $ do
     -- nsNextEphemeralAddr
     <*> pure nextEphemeralAddr
     -- nsBearerInfo
-    <*> initScript bearerInfoScript
+    <*> LazySTM.newTVar (initScript <$> script)
+    -- attenuationMap
+    <*> newTVar Map.empty
```

Comment:
This means we're actually never use the map.  Instead of passing `Script (Scipt BearerInfo)` to `newNetworkState` (and `withSnocket` as well), we should pass:

* `Map (ConnectionId peerAddr) (Script BearerInfo)`
* the default value for `Script BearerInfo`

Reply:
fixed!